drivers/usb/uhc/uhc_stm32.c

```diff
@@ -0,0 +1,1595 @@
+/*
```

Comment:
I have to complain again that it is not mentioned for which controller this driver is.
 
As far as I know there are two different ones that STM uses, Synopsys DWC USB 2.0 and something of their own (no idea if it can host). A look at `stm32cube/stm32h7xx/soc/stm32h723xx.h` confirms that this is a shim driver for the STM HAL DWC2 driver. We should have a native implementation for this (we will sooner or later anyway), similar to `drivers/usb/udc/udc_dwc2.c` (this also applies to other third party controllers or standardized designs used by many vendors, such as EHCI). So that we do not end up in the situation we have with `drivers/usb/device/usb_dc_stm32.c` and `drivers/usb/udc/udc_stm32.c`.

Reply:
> I have to complain again that it is not mentioned for which controller this driver is.

Well this is quite a new requirement that you come up with.

>  We should have a native implementation for this

Is that a new Zephyr requirement? Can you state doc for this ?

> So that we do not end up in the situation we have with `drivers/usb/device/usb_dc_stm32.c` and `drivers/usb/udc/udc_stm32.c`.

Can you elaborate ?

Reply:
Additional note on:

> We should have a native implementation for this

Why not. But this is not a valid reason to block a HAL based implementation which respect usual Zephyr criteria.

Reply:
See https://github.com/zephyrproject-rtos/zephyr/issues/61464#issuecomment-2302243662 for more details on the constraints of not using HAL.

As mentioned, STM32 USB IPs encompasses DWC2 but also 2 ST own USB IPs, and one of those actually has host capabilities. So current HAL based host STM32 driver will anyway be required as well for this IP.

Reply:
@jfischer-no I added a comment on top of the file for more explanation on the scope of the this driver. I hope it correspond to what you were expecting.
Also can you elaborate on what is problematic with the stm32 USB device driver ? One issue I saw is that parts with multiple USB cores are not supported but I tried to adresse such issue in this driver.

Also what is there any decision regarding DWC2 USB ? In any case part of the work done here can still be usefull for the stm32 parts not based on DWC2.